autoload/neomake/makers/ft/swift.vim

```diff
@@ -1,16 +1,24 @@
 " vim: ts=4 sw=4 et
 
 function! neomake#makers#ft#swift#EnabledMakers() abort
+    let list = ['swiftc']
     if !empty(s:get_swiftpm_config())
-        return ['swiftpm']
+        let list = ['swiftpm']
     endif
-    return ['swiftc']
+    if s:get_swiftlint_config()
+        let list = add(list, 'swiftlint')
+    endif
+    return list
 endfunction
 
 function! s:get_swiftpm_config() abort
     return neomake#utils#FindGlobFile('Package.swift')
 endfunction
 
+function! s:get_swiftlint_config() abort
+    return executable('swiftlint') && !empty(neomake#utils#FindGlobFile('.swiftlint.yml'))
+endfunction
```

Comment:
Isn't there also such a thing as a global config?

I think it makes sense to not require a local config for linters to be enabled.

Reply:
SwiftLint officially does not support global configs. It does have a default configuration which I think many wouldn't want to be enabled for every file they open. E.g. it marks every variable shorter than two letters as an error.
The convention is having a config file in your project or provide it via an argument.

I think the best solution would be to search for a config file then run `swiftlint`, check for ~/.swiftlint.yml if it exists run `swiftlintglobal` (with config provided).

Reply:
But wouldn't people not liking swiftlint then just not install it?

Reply:
I have it installed because some projects I work on use it, but others don't.
I cannot speak for everyone but I wouldn't want the default rules enabled and I'm not sure many are using a global config (there is a pretty old [issue](https://github.com/realm/SwiftLint/issues/606) requesting this, though)

Reply:
I've had a look at swift.vim.
There swiftlint is also only enabled when a `.swiftlint.yml` is present or the user requests it.
https://github.com/keith/swift.vim/blob/master/syntax_checkers/swift/swiftlint.vim#L14

What is your opinion? I would leave the current implementation as is. The user can always opt in and run/enable swiftlint for himself.